lib/ci-stack.ts

```diff
@@ -106,10 +116,24 @@ export class CIStack extends Stack {
     const certificateArn = Secret.fromSecretCompleteArn(this, 'certificateArn', importedArnSecretBucketValue.toString());
     const importedReloadPasswordSecretsArn = Fn.importValue(`${CIConfigStack.CASC_RELOAD_TOKEN_SECRET_EXPORT_VALUE}`);
     const listenerCertificate = ListenerCertificate.fromArn(certificateArn.secretValue.toString());
-    const agentNode = new AgentNodes();
-    const agentNodes: AgentNodeProps[] = [agentNode.AL2_X64, agentNode.AL2_X64_DOCKER_HOST, agentNode.AL2_X64_DOCKER_HOST_PERF_TEST,
-      agentNode.AL2_ARM64, agentNode.AL2_ARM64_DOCKER_HOST, agentNode.UBUNTU2004_X64, agentNode.UBUNTU2004_X64_DOCKER_BUILDER,
-      agentNode.MACOS12_X64_MULTI_HOST, agentNode.WINDOWS2019_X64];
+    const agentNode = new AgentNodes(this);
+
+    if (useProdAgents.toString() === 'true') {
+      // eslint-disable-next-line no-console
+      console.warn('Please note that if you have decided to use the provided production jenkins agents then '
+          + 'please make sure that you are deploying the stack in US-EAST-1 region as the AMIs used are only publicly '
+          + 'available in US-EAST-1 region. '
+          + 'If you want to deploy the stack in another region then please make sure you copy the public AMIs used '
+          + 'from us-east-1 region to your region of choice and update the code in agent-nodes.ts file accordingly. '
+          + 'If you do not copy the AMI in required region and update the code then the stack deployment in your desired '
+          + 'region will fail.');
+
+      this.agentNodes = [agentNode.AL2_X64, agentNode.AL2_X64_DOCKER_HOST, agentNode.AL2_X64_DOCKER_HOST_PERF_TEST,
+        agentNode.AL2_ARM64, agentNode.AL2_ARM64_DOCKER_HOST, agentNode.UBUNTU2004_X64, agentNode.UBUNTU2004_X64_DOCKER_BUILDER,
+        agentNode.MACOS12_X64_MULTI_HOST, agentNode.WINDOWS2019_X64];
+    } else {
+      this.agentNodes = [agentNode.AL2_X64_EXAMPLE_AGENT, agentNode.AL2_ARM64_EXAMPLE_AGENT];
```

Comment:
Can you just name them `AL2_X64` and `AL2_ARM64` remove example_agent as they can be actually used.

Reply:
I will replace `EXAMPLE` with `DEFAULT`.